akka-contrib/src/multi-jvm/scala/akka/contrib/pattern/ReliableProxySpec.scala

```diff
@@ -59,9 +59,22 @@ class ReliableProxySpec extends MultiNodeSpec(ReliableProxySpec) with STMultiNod
   }
 
   def stopProxy(): Unit = {
-    proxy ! FSM.UnsubscribeTransitionCallBack(testActor)
-    system stop proxy
-    expectMsgType[Terminated]
+    val oldProxy = proxy
+    oldProxy ! FSM.UnsubscribeTransitionCallBack(testActor)
+    system stop oldProxy
+    expectMsgPF(Duration.Undefined, "Terminated " + oldProxy) {
+      case Terminated(`oldProxy`) ⇒
+    }
+    // try to make really sure that the proxy actor name is available again
+    var id = 0
+    awaitCond({
+      id += 1
+      ActorSelection(proxy, Nil) ! Identify(id)
+      receiveOne(Duration.Undefined) match {
+        case ActorIdentity(id, None)             ⇒ true
+        case ActorIdentity(id, Some(`oldProxy`)) ⇒ false
+      }
```

Comment:
why not `expectMsg(ActorIdentity(id, None))`?

I don't think the `id` counter is important here